### ISSUE TITLE
Simplify top-level code to reduce cognitive complexity

### ISSUE THREAD

**Issue by msabramo** [chalk/chalk-cli#31]
![Screen Shot 2021-09-14 at 9 14 59 AM](https://user-images.githubusercontent.com/305268/133297092-8b0b22d3-b692-41a8-bee8-2bacf8dde988.png)

**Comment by Qix-** [comment]
Is this a problem reported with the linter? Or is this your own tooling?

Personally, I don't really understand such "limits". Code is code, sometimes there are longform functions. Breaking them out into individual functions only fragments control flow, usually making code harder to read (again, just my opinion).

**Comment by msabramo** [comment]
@Qix: I think this is arising from me having the [Visual Studio Code sonarlint extension](https://marketplace.visualstudio.com/items?itemName=SonarSource.sonarlint-vscode) installed.

So not everyone will encounter this warning and it doesn't show up in `npm run test` or CI checks, because those aren't using this sonarlint thingie. That said, eslint has the ability to do similar kinds of checks although I guess it's probably not turned on by xo.

As for the "limits", I'll add my own opinion about them and this may or may not affect your opinion, and either is fine - we all have our own stylistic preferences. But maybe it will be an interesting discussion, and hey because of the pandemic, I don't get to have as many of these technical discussions as I used to, so I'll wax philosophical for a bit... 😄 

I happen to use cyclomatic complexity checks (which I think is very similar to this "Cognitive Complexity" thing) pretty often in my day job projects and I find them helpful for pointing out places that could use some refactoring. I typically try to keep cyclomatic complexity of functions at maybe 10 or below .

I recognize that it's a machine using a dumb-simple heuristic to measure code and that it doesn't have human judgement so I was extremely skeptical of the usefulness of it at first.

But what I've found over the years is that very often the functions that this type of analysis point out as being overly complex are often the same functions that developers are afraid to touch and that are a common source of bugs. It's turned out be much more useful than I would've ever  imagined when I first encountered this stuff.

That said, this is a pretty simple piece of code and I don't see it being a huge source of bugs.

Although, I'll point out one more thing that was kind of interesting to me. It was only after I did this refactoring that I noticed from looking at the code that currently `--template` and `--stdin` are mutually exclusive.

In other words, you can do this:

```
$ node cli.js --template 'How about some {red.bold Unicorns & Rainbows}?'
How about some Unicorns & Rainbows?
```

but you **can't** do this:

```
$ echo 'How about some {red.bold Unicorns & Rainbows}?' | node cli.js --stdin --template
Input required
```

So I think there might be something to say about sometimes a bit of refactoring makes some issues visible that were harder to spot before.

But back to the issue at hand... Yeah, I think this warning is only coming from my editor. And so if @sindresorhus doesn't think this needs to be refactored either, then we can just close this issue.

**Comment by msabramo** [comment]
BTW, @Qix, I've really been enjoying collaborating with you and @sindresorhus the past few days! 😄

**Comment by msabramo** [comment]
FWIW, it seems that at least on [meow](https://www.npmjs.com/package/meow), `xo` is set up to detect complexity, although the max is set to 20:

```
abramowi at Marcs-MacBook-Pro-2 in ~/dev/git-repos/meow (main●●)
$ npm run test

> meow@10.1.1 test
> xo && ava && tsd


  index.js:1:1
  ⚠  102:14  Arrow function has a complexity of 25. Maximum allowed is 20.  complexity
...
```
